docs/llm/agui.md

```diff
@@ -0,0 +1,165 @@
+# AG-UI Protocol: Complete System Analysis
+
+## Overview
+
+AG-UI (Agent GUI) is a **thread-based conversational streaming protocol** that manages agent interactions through a hierarchical structure: **Room → Thread → Run → Events**.
+
+## Core Concepts
+
+### 1. Resource Hierarchy
+
+```
+Room (static config)
+  └── Thread (conversation container)
+        └── Run (single agent invocation)
+              └── Events (SSE stream)
+```
+
+### 2. API Flow Rules
+
+| Step | Action | Endpoint | Returns |
+|------|--------|----------|---------|
+| 1 | Create thread | `POST /rooms/{room_id}/agui` | Thread + **first Run UUID** |
+| 2 | Execute run | `POST /rooms/{room_id}/agui/{thread_id}/{run_id}` | SSE event stream |
+| 3 | Create subsequent run | `POST /rooms/{room_id}/agui/{thread_id}` | New Run UUID |
+| 4 | Execute subsequent run | `POST /rooms/{room_id}/agui/{thread_id}/{new_run_id}` | SSE event stream |
+
+**Critical constraints:**
+- A run can only be POSTed **once** (returns error if reused)
+- Thread creation includes the first run UUID to save a round-trip
+- The client manages message history and sends the full context with each run
+
+## Data Models
+
+### Thread
+```json
+{
+  "room_id": "string",
+  "thread_id": "uuid",
+  "runs": { "run_uuid": Run, ... },
+  "created": "datetime",
+  "metadata": { "name": "string", "description": "string" }
+}
+```
+
+### Run
+```json
+{
+  "thread_id": "uuid",
+  "run_id": "uuid",
+  "parent_run_id": "uuid | null",
+  "run_input": RunAgentInput,
+  "created": "datetime",
+  "events": [Event, ...],
+  "metadata": {}
+}
+```
+
+### RunAgentInput (request body for run execution)
```

Comment:
Do we need to point out that this one is isomorphic with the one defined in the AG-UI spec.